### PR TITLE
Run isort & black, automaticly check in CI

### LINKED ISSUE #296: Run `black` and `isort` automatically or as a hygiene check
It would be good to match the formatting that typeshed uses. They run `black` and `isort` automatically via a `pre-commit.ci` file. See https://github.com/python/typeshed/blob/main/CONTRIBUTING.md#code-formatting

### PR DESCRIPTION
**Pull request by Avasam**
Counterpart to #301, but with minimal config changes. So you can decide which you want first. There will be slight conflicts but I'll resolve them.
Closes #296

### CODE REVIEW

File: tests/requirements.txt
Comment (debonte) [inline tests/requirements.txt:1]: Similarly, here, I think we should keep isort/black separate from running tests. People can run black and isort however they see fit. For example, maybe as a VS Code plug-in.

**Review by debonte — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml
Comment (Avasam) [reply .github/workflows/test.yml:37]: I went with lint. Other possible names: checkers, linters, lint&format, linters&formatters, etc.

File: tests/requirements.txt
Comment (debonte) [inline tests/requirements.txt]: isort and black should also be removed from `run_tests.cmd` and `run_tests.sh`.

I'd be fine with having a separate script to run them, but automatically including them in all test runs seems wrong to me.

Comment (debonte) [inline tests/requirements.txt]: How about adding a `.vscode/extensions.json` file to suggest that people install the `ms-python.black-formatter` and `ms-python.isort` extensions?

```json
{
    // See https://go.microsoft.com/fwlink/?LinkId=827846
    // for the documentation about the extensions.json format
    "recommendations": [
        "ms-python.python",
        "ms-python.vscode-pylance",
        "ms-python.black-formatter",
        "ms-python.isort",
    ]
}
```

File: .github/workflows/test.yml
Comment (debonte) [reply .github/workflows/test.yml:45]: Well, black isn't a linter. Maybe isort is? I guess. Anyway, that's why I was suggesting "hygiene" since it covers a broad range of tool types.

**Review by debonte — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml
Comment (Avasam) [reply .github/workflows/test.yml:45]: I'm sorry, I somehow completely skipped over the fact that you actually suggested a name.

File: tests/requirements.txt
Comment (Avasam) [reply tests/requirements.txt]: I made a different script in #301 I can port it over or leave this PR w/o one for now and leave that discussion for that PR.

**Comment by Avasam** [comment]
Sorry for the late update! I've added mypy extension as well in preparation for #301

**Review by debonte — APPROVED** [review]

**Comment by Avasam** [comment]
@debonte Is this good to be merged?

**Comment by debonte** [comment]
@Avasam, sorry for the delay. I was probably, incorrectly, assuming that you could merge this yourself once it was approved.

Thanks for doing this! 💥

**Comment by Avasam** [comment]
No problem! You're welcome 😃

**Pull request merged**
